lib/web3/agentkit/action-providers/safe/index.ts

```diff
@@ -0,0 +1,109 @@
+import { z } from 'zod';
+
+const onchainAnalytics: OnchainAnalyticsProps = {
+    project: 'HELLO_WORLD_COMPUTER', // Required. Always use the same value for your project.
+    platform: 'WEB' // Optional
+};
+
+export class SafeActionProvider extends ActionProvider {
+
+    constructor() {
+        super("safe", []);
+    }
+
+    /**
+     * Creates a safe on the network.
+     *
+     * @param walletProvider - The wallet provider to create the safe from.
+     * @param args - The input arguments for the action.
+     * @returns A message containing the safe address.
+     */
+    @CreateAction({
+        name: "create_safe",
+        description: `
+      This tool will create a multisig safe wallet on the network. 
+      It takes the following inputs, both are addresses:
+        - owners: The addresses of the owners of the safe
+        - threshold: The number of owners required to sign a transaction
+      `,
+        schema: CreateSafeSchema,
+    })
+    async createSafe(
+        walletProvider: EvmWalletProvider,
+        args: z.infer<typeof CreateSafeSchema>
+    ): Promise<string> {
+        try {
+            const safeAccountConfig: SafeAccountConfig = {
+                owners: args.owners,
+                threshold: args.threshold
+                // ...
+            };
+
+            const predictedSafe: PredictedSafeProps = {
+                safeAccountConfig
+                // ...
+            };
+
+            const protocolKit = await Safe.init({
+                provider: baseSepolia.rpcUrls.default.http[0],//`https://base-sepolia.g.alchemy.com/v2/${process.env.ALCHEMY_API_KEY}`,
+                signer: walletProvider.getAddress(),
+                predictedSafe,
+                onchainAnalytics // Optional
+                // ...
+            });
+
+            const predictedSafeAddress = await protocolKit.getAddress();
+
+            const deploymentTransaction =
+                await protocolKit.createSafeDeploymentTransaction();
+
+
+            const client =
+                await protocolKit.getSafeProvider().getExternalSigner();
+
+            const tx = await client!.prepareTransactionRequest({
+                to: deploymentTransaction.to,
+                value: BigInt(deploymentTransaction.value),
+                data: deploymentTransaction.data as `0x${string}`,
+                chain: baseSepolia
+            });
+
+            const transactionHash = await walletProvider.sendTransaction(tx);
+
+            await waitForTransactionReceipt(
+                client!,
+                { hash: transactionHash }
+            );
+
+            const newProtocolKit = await protocolKit.connect({
+                safeAddress: predictedSafeAddress
+            });
+
+            const deployedSafeAddress = await newProtocolKit.getAddress();
+
+            return `Safe created at ${deployedSafeAddress} with threshold ${args.threshold} and owners ${args.owners.join(', ')}`;
```

Comment:
Rather than a string, can we return typed data, something like:
{
  safeAddress: 0x
  transactionHash: 0x
  threshold: 1
  owners: [0x,0x]
}
So we can more easily render it nicely on the frontend